File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/python/F14.P1.py
from dataclasses import dataclass


# BEGIN PROBE F14.P1
@dataclass
class Circle:
    r: float


@dataclass
class Rect:
    w: float
    h: float


Shape = Circle | Rect
s: Shape = Circle(2.0)
# END PROBE F14.P1

print(s.r)

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/typescript/F14.P1.ts
function newShape() {
// BEGIN PROBE F14.P1
interface Circle { kind: "Circle"; r: number }
interface Rect { kind: "Rect"; w: number; h: number }
type Shape = Circle | Rect
const s: Shape = { kind: "Circle", r: 2.0 }
// END PROBE F14.P1
return s
}

console.log(JSON.stringify(newShape()))
